### PR TITLE
nix: build pg-utils with nix & upload to GCS

### PR DESCRIPTION
**Pull request by Strum355**
Lets get rid of PG_UTILS_PATH! :allthethings: :tada: 

## Test plan

CI and a _lot_ of  `nix build .#pg-utils && ldd ./result/bin/pg_dump`
also tested e2e with the following diff: https://github.com/sourcegraph/sourcegraph/commit/61e0eb0d556dbfbcdae7a0c2cac1f5dd5cb85553

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -43,21 +43,23 @@
...
+            pg-utils = pkgs.callPackage ./dev/nix/pg-utils.nix { };
           };
 
           # We use pkgsShell (not pkgsAll) intentionally to avoid doing extra work of
           # building static comby/universal-ctags in our development
           # environments.
           devShells.default = pkgsShell.callPackage ./shell.nix { };
 
-          formatter = pkgs.nixpkgs-fmt;
+          # formatter = pkgs.nixpkgs-fmt;
```
Comment (jhchabran) [inline flake.nix:54]: Why the comment here?

**Review by jhchabran — COMMENTED** [review]
Nice, left a small comment! Eager to get this up and running, PG_UTILS have been creating a few issues every now and then 🙏 

I suppose you intend to remove the `sg` part in another PR?

File: flake.nix (hunk 1 above)
Comment (Strum355) [reply flake.nix:54]: I can probably revert this and just disable format-on-save in my editor, it was slowing the save loop

**Comment by Strum355** [comment]
> Nice, left a small comment! Eager to get this up and running, PG_UTILS have been creating a few issues every now and then 🙏
> 
> I suppose you intend to remove the `sg` part in another PR?

Ill be doing a second PR to add the bazel side of things, I guess we have 3 options:
1. remove sg side of things in this PR
2. remove sg side of things in the bazel PR
3. remove sg side of things in a final 3rd PR
Any preference? @jhchabran

**Comment by jhchabran** [comment]
> > Nice, left a small comment! Eager to get this up and running, PG_UTILS have been creating a few issues every now and then 🙏
> > I suppose you intend to remove the `sg` part in another PR?
> 
> Ill be doing a second PR to add the bazel side of things, I guess we have 3 options:
> 
>     1. remove sg side of things in this PR
> 
>     2. remove sg side of things in the bazel PR
> 
>     3. remove sg side of things in a final 3rd PR
>        Any preference? @jhchabran

none, just do what's the most convenient for you :)

File: dev/nix/pg-utils.nix [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    ${patchelf}/bin/patchelf --print-needed $out/bin/pg_dump | \
+      ${gnugrep}/bin/grep -v 'libc.so.6' && echo 'unexpected dynamic library dependency found, only libc is expected' && exit 1 \
+      || exit 0
+  '';
+
+  # update linker to not point to the nix one, but to one that will work on
+  # most other distros such as Ubuntu
+  postFixup = lib.optionalString stdenv.isLinux ''
+    for bin in $out/bin/*; do
+      ${patchelf}/bin/patchelf --set-interpreter /lib64/ld-linux-x86-64.so.2 "$bin"
```
Comment (burmudar) [inline dev/nix/pg-utils.nix:71]: :galaxybrain:

**Review by burmudar — APPROVED** [review]
The nix is putting everything together is
gif

**Pull request merged**